### DISCUSSION TITLE
Thread ESP32 Border Router - Failed to send IPv6 UDP msg

### DISCUSSION THREAD

**Discussion by hertugweile** [openthread/openthread#9802] [Q&A]
Hi, I have this setup with two ESP's, connected together like this - https://github.com/espressif/esp-thread-br - the first one is a C6, the second one is a H2. 

The problem is that I receive a network error on the C6, every time they communicate - or so it seems:
```
I(124004) OPENTHREAD:[N] MeshForwarder-: Failed to send IPv6 UDP msg, len:90, chksum:d57b, ecn:no, to:0xffff, sec:no, error:ChannelAccessFailure, prio:net
I(124014) OPENTHREAD:[N] MeshForwarder-:     src:[fe80:0:0:0:cc51:1e3e:7fdc:feb9]:19788
I(124014) OPENTHREAD:[N] MeshForwarder-:     dst:[ff02:0:0:0:0:0:0:1]:19788
```

I'm unsure if the problem lies within my internet router or the ESP32. Searching on the error returns very few results. I have also flashed the C6 as a stand-alone unit without the H2 and changed the settings in order to enable this functionality, but I get the same error.

Any ideas?

**Comment by chshu** [comment]
The `ChannelAccessFailure` error indicates the noisy around is high, resulting in transmission failures due to Clear Channel Assessment (CCA) issues.

To assess the actual noise level, please flash the [ieee802154_cli](https://github.com/espressif/esp-idf/tree/master/examples/ieee802154/ieee802154_cli) example on either C6 or H2. And use command `ed -d 100` to check the actual noise level.

If the noise level looks good, but the issue persists consistently (rather than occasionally), kindly provide the IDF commit ID and details of your setup, we will reproduce it on our end.

**Comment by hertugweile** [comment]
I tried to move it away from everything - and put the h2 in the other end of the breadboard, facing the other direction, but still no luck. I then flashed the H2 with the example and it shows -102 dB.
Under all circumstances; I've given up on this and have ordered a couple of Espressif ESP Thread BR-Zigbee GW's, as I would like to use the GUI as well.

**Reply by chshu** [reply]
The [example](https://github.com/espressif/esp-thread-br/tree/main/examples/basic_thread_border_router) in esp-thread-br repo works as is on [ESP Thread BR Board](https://docs.espressif.com/projects/esp-thread-br/en/latest/hardware_platforms.html#wi-fi-based-thread-border-router), let me know in case you encounter any issue with it.

In order to setup the Thread BR with C6 + H2, here are the steps for your reference:
- build and flash [ot_rcp](https://github.com/espressif/esp-idf/tree/master/examples/openthread/ot_rcp) to ESP32-H2 devkit

- use [ot_br](https://github.com/espressif/esp-idf/tree/master/examples/openthread/ot_br) example in esp-idf

    - idf.py set-target esp32c6
    - configure `OPENTHREAD_RADIO_TYPE` to "Connect via UART" (note: C6 has 802.15.4, so the default is "Native 15.4 radio")
    - update the correct [C6 pin numbers](https://github.com/espressif/esp-idf/blob/master/examples/openthread/ot_br/main/esp_ot_config.h#L40) connected to H2's TX and RX
    - build and flash the example to ESP32-C6 devkit

- connect C6 and H2 with three wires (TX, RX, GND), be sure to close the H2 console monitor, since the UART0 is used for RCP UART

Then there should be some output as below:

```
I (611M�I (621) OPENTHREAD: spinel UART interface initialization completed
I (621) main_task: Returned from app_main()
I(621) OPENTHREAD:[I] P-RadioSpinel-: RCP reset: RESET_POWER_ON
I(631) OPENTHREAD:[I] P-RadioSpinel-: Software reset RCP successfully
I(671) OPENTHREAD:[I] ChildSupervsn-: Timeout: 0 -> 190
I (671) OPENTHREAD: OpenThread attached to netif
> scan

| PAN  | MAC Address      | Ch | dBm | LQI |
+------+------------------+----+-----+-----+
| b335 | 92eacce0b84594c1 | 11 | -63 |   9 |
| 1a65 | e6bd350d91e0c183 | 13 | -45 |   9 |
| 1a65 | 9eed1d95cf1004b3 | 13 | -44 |   8 |
| 35e9 | 8122334455667705 | 15 | -55 |  10 |
| e2b2 | 0e5ba4c40bf0b6be | 16 | -49 |  10 |
| e2b2 | 360d34996a2c9e34 | 16 | -43 |  10 |
| 4d83 | ba9741cd44273f44 | 18 | -45 |   9 |
| 472f | da75c0910ffbb2a7 | 24 | -53 |  10 |
| 8df9 | c60e461cbc93936c | 25 | -58 |   7 |
| 45f7 | 7e062787e985c640 | 25 | -57 |   9 |
Done
>
```
